### DISCUSSION TITLE
mitosis-packager

### DISCUSSION THREAD

**Discussion by moxley01** [BuilderIO/jsx-lite#1267] [Show and tell]
Hey guys,

I've been exploring the Mitosis framework and found it to be a promising tool for project development. I wanted to share some build tooling we've created to help streamline the project setup process and integrate Cypress for end-to-end testing.

It is quite inspired by what `create-react-app` has done for React, allowing you to quickly scaffold a new project.

Likewise, by calling `npx create-multi-lib <your-lib-name>` a CLI tool prompts you to select some target frameworks, and scaffold a project that uses Vite to bundle Mitosis output into minified, publishable packages, with the same set of Cypress E2E testing assertions running on each.

Here is the repo: https://github.com/magicbell-io/mitosis-packager

Link to demo video:
https://www.youtube.com/watch?v=m1vCRogZ6HA

**Comment by samijaber** [comment]
This is awesome! I love to see what the community sees as important in terms of streamlining the project setup process.

I've chosen to keep Mitosis build-agnostic for now due to the huge variety in possible setups (yarn/pnpm/npm workspaces, cypress vs playwright, etc.), but having something like this out there is huge. You definitely have the right idea, the project setup is similar in a lot of ways to what we do for the Builder SDKs: https://github.com/BuilderIO/builder/tree/main/packages/sdks

Having your packager handle more important outputs (Qwik, React Native, etc.) will be huge! Let me know if you need any pointers on how to move it forward. 😄
